cst438_register/src/main/java/com/cst438/controller/ScheduleController.java

```diff
@@ -43,28 +43,116 @@ public class ScheduleController {
 	GradebookService gradebookService;
 	
 	
+	/* create REST API to add or change a student record.
+	admin can add a student to the system. By inputting the email and name.*/  
+   @PostMapping("/addStudent")
+   @Transactional
+   public ScheduleDTO.StudentDTO addStudent( @RequestBody ScheduleDTO.StudentDTO studentDTO) 
+   { 
+   String student_email = studentDTO.email;   // student's email 
+   String name = studentDTO.name;
+   Student student = studentRepository.findByEmail(student_email);
+      if (student != null) {
+         System.out.println("/student email already in database. "+student_email);
+         throw  new ResponseStatusException( HttpStatus.CONFLICT, "Email already in use. " );
+      }
+      else {
+         System.out.println("/addStudent called.");
+         //IF STUDENT IS NOT ENTERED ALREADY, THE NAME IS VALID AND EMAIL IS VALID
+         if (student == null && name!= null && student_email != null) {
+            student = new Student();
+            //INIT STUDENT OBJECT//
+            student.setName(name);
+            student.setEmail(student_email);
+            Student savedStudent = studentRepository.save(student);
+            ScheduleDTO.StudentDTO result = createStudentDTO(savedStudent);
+            return result;
+         } else {
+            throw  new ResponseStatusException( HttpStatus.BAD_REQUEST, "Response msg"+studentDTO.name);
+            }
+      }
+   }
+	
+   //Admin can put student registration on HOLD.
+	@PostMapping("/addStudentHold")
```

Comment:
```suggestion
	@PostMapping("/schedule/student/hold")
```

I would break up the URL query to be more precise.